### ISSUE TITLE
✨Feature request:  Engine manager supports all collapse/expand

### ISSUE THREAD

**Issue by dnknn** [ssborbis/ContextSearch-web-ext#319]
- `chrome-extension://ddippghibegbgpjcaaijbacfhjjeafjh/options.html#engines`

- For the folders: have a `li.treeNode .collapse`

---

- ## 💥 **Current behavior:**
　　　🛇. Only click one by one to expand/collapse
　　　🛇. Unable to implement one operation to expand/collapse all

- ## 💡 **Implementation plan:**
　　　1. Add a `button`, the user can click it to expand/collapse all
　　　2. Clicking `li.treeNode .collapse` is current behavior, then add: 
Holding `Ctrl/Shift/Ctrl+Shift` + 　Clicking `li.treeNode .collapse` is to expand/collapse all !

- ### Best reference:  gifs demo is provided!

[GitHub Toggle Expanders](https://greasyfork.org/scripts/23303) 　　([github](https://github.com/Mottie/GitHub-userscripts/wiki/GitHub-toggle-expanders))

> Use Shift/Ctrl + Click on an expander to toggle all expanders

https://raw.githubusercontent.com/Mottie/GitHub-userscripts/master/images/github-toggle-expanders.gif

**Comment by ssborbis** [comment]
Newer code has an expand/collapse-all button at the top of the tree. I'll look into the modifier keys for collapsing branches. Do you have that many nested folders?

**Comment by dnknn** [comment]
> Do you have that many nested folders?

Not many at present, but with more engines, folders are naturally needed for classification and processing.

**Comment by dnknn** [comment]
> Newer code has an expand/collapse-all button at the top of the tree.

Because there are already UI buttons, it is more straightforward and simple. 👇
![1111](https://user-images.githubusercontent.com/10995302/110049970-d9a10000-7d8d-11eb-90fe-97341c31f40c.png)

> I'll look into the modifier keys for collapsing branches. 

So the modifier keys for collapsing branches, there is no need to study it anymore.
Even if we support the modifier keys for collapsing branches, it will is purely furnishings/decoration only, 99% is definitely the first choice ![image](https://user-images.githubusercontent.com/10995302/110050184-5c29bf80-7d8e-11eb-96e2-48e062ab1417.png)

**Comment by dnknn** [comment]
but there is a situation, when the scroll bar is scrolled down, this ![image](https://user-images.githubusercontent.com/10995302/110050539-11f50e00-7d8f-11eb-8c28-820592f072ed.png) button will not be visible....it cannot be clicked...

@ssborbis  Can you make CSS changes to make it fixed ?❓	

```css
#managerContainer>.collapse	{position:absolute}
```

　　　👆 change to 👇

```css
#managerContainer>.collapse	{position:fixed}
```

**Comment by ssborbis** [comment]
Got it. 

Also, something you might like. I added a search bar to the top of the search engines manager to filter engines.

![Peek 2021-03-06 07-30](https://user-images.githubusercontent.com/24310997/110208610-d5d4b280-7e4d-11eb-8298-34a624f02990.gif)

**Comment by dnknn** [comment]
> I added a search bar to the top of the search engines manager to filter engines.

Very great! I have suggestions!

## Support `Ctrl+F` to focus this search bar ! 

>just like the manager  of [Tampermonkey](https://chrome.google.com/webstore/detail/dhdgffkkebhmkfjojejmpbldmpobfkfo) or [Stylus](https://chrome.google.com/webstore/detail/clngdbkpkpeebahjckkjfobafhncgmne).


----

## Non-main content does not occupy visual space.

![image](https://user-images.githubusercontent.com/10995302/110209208-1ec84d80-7ec6-11eb-9c49-a704ecccacbb.png)


The search engine manager page should maximize the display of the content, so it is recommended to remove the non-main content such as the top `Header`..etc.

In other words, we can use workarounds to implement content such as help information, but the principle is non-main content will not take up additional display space.

**Comment by dnknn** [comment]
> I added a search bar to the top of the search engines manager to filter engines.

Very great! I have suggestions!

## Support `Ctrl+F` to focus this search bar ! 

>just like the manager  of [Tampermonkey](https://chrome.google.com/webstore/detail/dhdgffkkebhmkfjojejmpbldmpobfkfo) or [Stylus](https://chrome.google.com/webstore/detail/clngdbkpkpeebahjckkjfobafhncgmne).

## `Esc` to clear the input in the search bar (⚠when the focus is on the search bar)

----

## Non-main content does not occupy visual space.

![image](https://user-images.githubusercontent.com/10995302/110209208-1ec84d80-7ec6-11eb-9c49-a704ecccacbb.png)


The search engine manager page should maximize the display of the content, so it is recommended to remove the non-main content such as the top `Header`..etc.

In other words, we can use workarounds to implement content such as help information, but the principle is non-main content will not take up additional display space.
